infra/frontend/service/main.tf

```diff
@@ -104,9 +108,11 @@ module "service" {
   vpc_id                = data.aws_vpc.default.id
   subnet_ids            = data.aws_subnets.default.ids
   enable_autoscaling    = module.app_config.enable_autoscaling
-  # TODO: Reroute SSL to production environment
-  # To temporarily avoid issues in production & Terratest: Check if environment is "dev" and workspace is "default"
-  cert_arn = var.environment_name == "dev" && terraform.workspace == "default" ? data.aws_acm_certificate.frontend_cert.arn : null
+  cert_arn = (
+    terraform.workspace != "default" ? null :
+    var.environment_name == "prod" ? data.aws_acm_certificate.frontend_prod_cert.arn :
+    data.aws_acm_certificate.frontend_dev_cert.arn
+  )
```

Comment:
```suggestion
  # Temporary to avoid issues if Terratest if workspace is default and pass correct environment cert
  cert_arn = (
    terraform.workspace != "default" ? null :
    (var.environment_name == "prod" ? data.aws_acm_certificate.frontend_prod_cert.arn :
    data.aws_acm_certificate.frontend_dev_cert.arn)
  )
```

Reply:
I take responsibility for this initial hack and added a comment that this should be temporary. A couple things:

1. Do we even need SSL on dev? I'm ok leaving it, just wondering.
2. An idea on how to fix this logic: 
Create locals for domains, something like:

```
  cert_domains = {
    dev  = "dev.yourdomain.com"
    prod = "prod.yourdomain.com"
  }
```
- Move cert data blocks into `/app-config`, can be done in a loop if you want
```
data "aws_acm_certificate" "cert" {
  for_each = toset(local.environments)
  
  domain = local.cert_domains[each.key]
  ...
}
```
- Add cert arn to env_config
`cert_arn  = data.aws_acm_certificate.cert[each.key].arn`
  
I didn't try this out so might be syntax errors, but something like this.

Reply:
I take responsibility for this initial hack and added a comment that this should be temporary. A couple things:

1. Do we even need SSL on dev? I'm ok leaving it, just wondering.
2. An idea on how to fix this logic: 
Create locals for domains in `app_config`, something like:

```
  cert_domains = {
    dev  = "beta.grants.gov"
    prod = "simpler.grants.gov"
  }
```
- Move cert data blocks into `app-config`, can be done in a loop
```
data "aws_acm_certificate" "cert" {
  for_each = toset(local.environments)
  
  domain = local.cert_domains[each.key]
  ...
}
```
- Add cert arn to `env_config` in `app_config`
`cert_arn  = data.aws_acm_certificate.cert[each.key].arn`
  
I didn't try this out so might be syntax errors, but something like this. This doesn't address Terratest and workspace, but at least it's moving environment logic out.

Reply:
EDIT: Can't retrieve certs based on tag names, so changed to use domain. Thank you for pointing that out, Sammy

Wait, even simpler, didn't see this earlier. Pass domain names in through `app_config`.

In `app_config` `locals`:

```
 cert_domains = {
    dev  = "beta.grants.gov"
    prod = "simpler.grants.gov"
  }
``` 

In `app_config` `env_config`:

```
domain_name = local.cert_domains[each.key]
```

Instead of this:

```
data "aws_acm_certificate" "frontend_dev_cert" {
  domain = "beta.grants.gov"
}

data "aws_acm_certificate" "frontend_prod_cert" {
  domain = "simpler.grants.gov"
}
```

Do this to retrieve cert:

```
data "aws_acm_certificate" "cert" {
  domain = var.domain_name
}
```

Then just pass `aws_acm_certificate` to service module and no need for ternary for environments. I like this way better cause simpler is good and I wasn't sure how I felt about data blocks in `app_config`

Reply:
Ok... what if... we just put it in `dev.tfvars` and `prod.tfvars` 😆 

For example, in `dev.tfvars`
`domain_name = "beta.grants.gov"`

Then in service use it as `var.domain_name`.